File: client/scalaz/src/main/scala/org/openapitools/client/api/CdashigDomainField.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CdashigDomainField._

case class CdashigDomainField (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
definition: Option[String],
questionText: Option[String],
prompt: Option[String],
completionInstructions: Option[String],
implementationNotes: Option[String],
simpleDatatype: Option[String],
mappingInstructions: Option[String],
core: Option[String],
links: Option[CdashigDomainFieldLinks])

object CdashigDomainField {
  import DateTimeCodecs._

  implicit val CdashigDomainFieldCodecJson: CodecJson[CdashigDomainField] = CodecJson.derive[CdashigDomainField]
  implicit val CdashigDomainFieldDecoder: EntityDecoder[CdashigDomainField] = jsonOf[CdashigDomainField]
  implicit val CdashigDomainFieldEncoder: EntityEncoder[CdashigDomainField] = jsonEncoderOf[CdashigDomainField]
}

File: client/clojure/src/cdisc_library_api/specs/cdashig_domain_field.clj
(ns cdisc-library-api.specs.cdashig-domain-field
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.cdashig-domain-field-links :refer :all]
            )
  (:import (java.io File)))


(def cdashig-domain-field-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :definition) string?
   (ds/opt :questionText) string?
   (ds/opt :prompt) string?
   (ds/opt :completionInstructions) string?
   (ds/opt :implementationNotes) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :mappingInstructions) string?
   (ds/opt :core) string?
   (ds/opt :_links) cdashig-domain-field-links-spec
   })

(def cdashig-domain-field-spec
  (ds/spec
    {:name ::cdashig-domain-field
     :spec cdashig-domain-field-data}))
